src/firebaseDataProvider.ts

```diff
@@ -150,6 +150,19 @@ class FirebaseClient {
     params: IParamsCreate
   ): Promise<IResponseCreate> {
     const r = await this.tryGetResource(resourceName);
+    const newId = params.data[CREATE_WITHOUT_AUTOMATIC_ID_KEY];
+    if (newId) {
+      const data = {
+        ...params.data
+      };
+      delete data[CREATE_WITHOUT_AUTOMATIC_ID_KEY];
+      await r.collection.doc(newId).set(data, { merge: true });
+      return {
+        data: {
+          id: newId
+        }
```

Comment:
Please put `id` in extra field

Reply:
Here's the definition of `IResponseCreate`:

```
interface IResponseCreate {
  data: {};
}
```